src/manual/media-recorder/js/main.js

```diff
@@ -0,0 +1,441 @@
+/*
+*  Copyright (c) 2016 The WebRTC project authors. All Rights Reserved.
+*
+*  Use of this source code is governed by a BSD-style license
+*  that can be found in the LICENSE file in the root of the source
+*  tree.
+*/
+
+'use strict';
+
+function RecorderTest() {
+  this.theLocalStream = null;
+  this.theRemoteStream = null;
+  this.theRecorder = null;
+  this.eventsMonitored = null;
+  // Note: ondataavailable is not in the list below because it is necessary to
+  // have it always on if we ever want to playbabk/download the recording in a
+  // consistent manner.
+  this.RECORDER_EVENTS = ['onstart', 'onpause', 'onresume', 'onstop',
+      'onerror'];
+  this.eventCounter = 0;
+  this.recordedData = {};
+  this.timeDataReceived = null;
+}
+
+RecorderTest.prototype = {
+  // This function sets up the different event handlers necessary to run the
+  // ui.
+  setup: function() {
+    this.onButtonClick('GetUserMedia', this.getUserMedia.bind(this));
+    this.onButtonClick('CreatePeerConnection', this.createPeerConnection.bind(
+        this));
+    this.onButtonClick('CreateRecorder', this.createRecorder.bind(this));
+    this.onButtonClick('Start', this.startRecorder.bind(this));
+    this.onButtonClick('Pause', this.pauseRecorder.bind(this));
+    this.onButtonClick('Resume', this.resumeRecorder.bind(this));
+    this.onButtonClick('Stop', this.stopRecorder.bind(this));
+    this.onButtonClick('RequestData', this.requestDataRecorder.bind(this));
+    this.onButtonClick('Playback', this.playback.bind(this));
+    this.onButtonClick('Download', this.download.bind(this));
+    var monitorSelect = document.getElementById('monitorEvents');
+    monitorSelect.onchange = this.monitorRecorderEvents.bind(this, event);
+  },
+
+  // This function creates a button onClick event handler.
+  onButtonClick: function(buttonName, callBack) {
```

Comment:
Not a fan of wrapping the event listener (or any standard method) method like this.

What about keeping it simple albeit more lines of codes:
```
var getUserMedia = document.querySelector('#GetUserMedia');
getUserMedia.addEventListener('click', this.getUserMedia.bind(this));
```